src/librustc_typeck/check/_match.rs

```diff
@@ -266,6 +330,7 @@ impl<'a, 'tcx> FnCtxt<'a, 'tcx> {
         else_expr: &'tcx hir::Expr<'tcx>,
         then_ty: Ty<'tcx>,
         else_ty: Ty<'tcx>,
+        suggest_box: Option<Span>,
```

Comment:
`suggest_box` being an `Option` instead of a bool seems somewhat weird to me.

Not sure what's a better name here though :thinking: maybe `opt_suggest_box_span` or something like this.

Reply:
It was originally a bool, which is why it is weird 😅